.buildkite/steps/publish.sh

```diff
@@ -32,15 +24,19 @@ echo "--- Building :cloudformation: CloudFormation templates"
 make build/aws-stack.yml
 
 echo "--- Uploading :cloudformation: CloudFormation templates"
-
-# Publish the top-level mappings only on when we see the most recent tag on master
-if is_latest_tag ; then
-  if ! is_prerelease_tag ; then
-    s3_upload_templates "latest/"
-  fi
+trunk="origin/$BUILDKITE_PIPELINE_DEFAULT_BRANCH"
+latest_tag=$(git describe --tags --abbrev=0 --match='v*' "$trunk")
+# Pre-release tags are those that have a hyphen in them, e.g. v1.0.0-rc1
+latest_stable_tag=$(git describe --tags --abbrev=0 --match='v*' --exclude='*-*' "$trunk")
+
+# Only publish to 'latest' (and the empty prefix) if this tag is the latest stable tag.
+if [[ $BUILDKITE_TAG == "$latest_stable_tag" ]]; then
+  s3_upload_templates "latest/"
   s3_upload_templates
+elif [[ $BUILDKITE_TAG == "$latest_tag" ]]; then
+  echo "Skipping publishing latest, although $BUILDKITE_TAG matchs $latest_tag it does not doesn't match $latest_stable_tag"
```

Comment:
Just the things I remember from https://google.github.io/styleguide/shellguide.html (which I should probably re-read)

(there's also a spelling fix in here)

```suggestion
trunk="origin/${BUILDKITE_PIPELINE_DEFAULT_BRANCH}"
latest_tag="$(git describe --tags --abbrev=0 --match='v*' "$trunk")"
# Pre-release tags are those that have a hyphen in them, e.g. v1.0.0-rc1
latest_stable_tag="$(git describe --tags --abbrev=0 --match='v*' --exclude='*-*' "$trunk")"

# Only publish to 'latest' (and the empty prefix) if this tag is the latest stable tag.
if [[ "${BUILDKITE_TAG}" == "${latest_stable_tag}" ]]; then
  s3_upload_templates "latest/"
  s3_upload_templates
elif [[ "${BUILDKITE_TAG}" == "${latest_tag}" ]]; then
  echo "Skipping publishing latest, although ${BUILDKITE_TAG} matches ${latest_tag} it does not doesn't match ${latest_stable_tag}"
```